client/lib/core/widgets/media_settings_widget.dart

```diff
@@ -0,0 +1,475 @@
+          const SizedBox(height: 24),
+          Text(
+            'Video Input Device',
+            style: context.theme.textTheme.titleMedium,
+          ),
+          _mediaService.videoInputs.isEmpty
+              ? Padding(
+                  padding: const EdgeInsets.only(top: 16.0),
+                  child: Text(
+                    'No video devices available. Please check permissions.',
+                    style: context.theme.textTheme.bodyMedium!.copyWith(
+                      color: context.theme.colorScheme.onSurfaceVariant,
+                    ),
+                  ),
+                )
+              : DropdownButton<String>(
+                  // Prevent dropdown errors by first checking that the selected
+                  // device still exists in available inputs.
+                  value: _mediaService.videoInputs.any(
+                    (device) =>
+                        device.deviceId == _mediaService.selectedVideoInputId,
+                  )
+                      ? _mediaService.selectedVideoInputId
+                      : null,
+                  items: _mediaService.videoInputs.map((device) {
+                    return DropdownMenuItem<String>(
+                      value: device.deviceId,
+                      child: Text(
+                        device.label!,
+                        style: context.theme.textTheme.titleMedium,
+                      ),
+                    );
+                  }).toList(),
+                  selectedItemBuilder: (BuildContext context) {
+                    return _mediaService.videoInputs.map<Widget>((device) {
+                      return Container(
+                        alignment: Alignment.centerLeft,
+                        constraints: BoxConstraints(
+                          maxWidth: 320,
+                        ),
+                        // 24px for the dropdown arrow
+                        width: MediaQuery.of(context).size.width -
+                            _kTotalDialogContentPadding -
+                            24,
+                        child: Text(
+                          device.label!,
+                          style: context.theme.textTheme.titleMedium,
+                          overflow: TextOverflow.ellipsis,
+                          softWrap: true,
+                        ),
+                      );
+                    }).toList();
+                  },
+                  onChanged: (val) async {
+                    if (val != null) {
+                      setState(() {
+                        isLoading = true;
+                      });
+                      await _mediaService.selectVideoDevice(
+                        deviceId: val,
+                        shouldUpdatePreview: widget.shouldShowVideoPreview,
+                      );
+                      await updatePreview();
+                      setState(() {
+                        isLoading = false;
+                      });
+                    }
```

Comment:
Same as my comment above, I think doing `if (val == null) return;` and then following it up with the rest of the code to be executed otherwise is more readable.